ivadomed/transforms.py

```diff
@@ -287,6 +294,50 @@ def __call__(self, sample, metadata=None):
         return data_out, metadata
 
 
+class VertebralSplitting(ImedTransform):
+    """Create an intervertebral disk masks for pose estimation model .
+    NB: This transform is not multichannel capable. It is meant to be used
+    to *create* a multichannel image from a single-channel image.
+    """
+
+    def __init__(self, max_joint=11):
+        self.max_joint = max_joint
+
+    def undo_transform(self, sample, metadata=None):
+        return sample, metadata
+
+    def get_posedata(self, msk, num_ch):
+        ys = msk.shape
+        ys_ch = [np.zeros([ys[0], ys[1]])] * num_ch
+        msk_uint = np.uint8(np.where(msk>0.2, 1, 0))
+        
+        labels_im, num_labels = measure.label(msk_uint, background=0, return_num=True)
```

Comment:
> My gut feeling is that this was done for displaying images in the correct orientation for generating figures for the paper, but this is not necessary for the actual function.

I've spoken with Reza, and the reason the images were rotated was because of the [`measure.label()`](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.ndimage.measurements.label.html) connected components labeling algorithm used to split the ground truth image. This algorithm uses the Y coordinate to number the segments, so images like this become problematic, since the Y coordinates are not monotonic:

![image](https://user-images.githubusercontent.com/16181459/124973127-25440e80-dff9-11eb-86bb-696910ce43e6.png)

But, as you've pointed out @jcohenadad  the _data array itself_ was flipped during preprocessing. Instead, if I'm not mistaken, the proper way to make sure the images are in the correct orientation would have been to use the header orientation metadata (and leave the array itself alone).